pkg/controller/core/workload_controller_test.go

```diff
@@ -1641,6 +1641,14 @@ func TestReconcile(t *testing.T) {
 					Message: "The LocalQueue is stopped",
 				}).
 				Obj(),
+			wantEvents: []utiltesting.EventRecord{
+				{
+					Key:       types.NamespacedName{Name: "wl", Namespace: "ns"},
+					EventType: corev1.EventTypeNormal,
+					Reason:    "EvictedDueToLocalQueueStopped",
+					Message:   "The LocalQueue is stopped",
+				},
+			},
```

Comment:
@mimowo you probably want to double check that this updated test is correct.

Reply:
+1. The reason is that inside `reconcileOnLocalQueueActiveState` we called the metrics bump skipping the event emission. I think the fact the event was not emitted for eviction in this scenario could be called a consistency bug, because we would emit the event in all other eviction scenarios. 

If you extract most of the PR (except adding the new metric), then I merge it as bugfix and cherry-pick onto 0.12 and 0.11. It is not really a must, because it is a minor bug no-one called yet for.  OTOH, someone may call it in the future. Are you up to?